classes/Search.php

```diff
@@ -182,15 +182,17 @@ public static function sanitize($string, $id_lang, $indexation = false, $iso_cod
             $words = explode(' ', $string);
             $processed_words = [];
             // search for aliases for each word of the query
-            foreach ($words as $word) {
-                $alias = new Alias(null, $word);
-                if (Validate::isLoadedObject($alias)) {
-                    $processed_words[] = $alias->search;
-                } else {
-                    $processed_words[] = $word;
-                }
+            $aliases = Db::getInstance(_PS_USE_SQL_SLAVE_)->executeS('
+				SELECT a.alias, a.search
+				FROM `' . _DB_PREFIX_ . 'alias` a
+				WHERE \'' . pSQL($string) . '\' LIKE CONCAT(\'%\', alias, \'%\') AND `active` = 1');
```

Comment:
```suggestion
				WHERE \'' . pSQL($string) . '\' REGEXP CONCAT(\'(^|[[:space:]]|[[:<:]])\', alias, \'([[:space:]]|[[:>:]]|$)\') AND `active` = 1');
```

Because if you have a lot of word, the search is not looking for boundaries.
Example:
You have this:
![image](https://user-images.githubusercontent.com/1462701/102865806-95f95400-4436-11eb-861d-c01f161fca7b.png)

With LIKE and CONCAT:
![image](https://user-images.githubusercontent.com/1462701/102866071-fd170880-4436-11eb-8caa-63b184d3b828.png)


With word boundaries (compatible MySQL 5.6 :sob: )
![image](https://user-images.githubusercontent.com/1462701/102866123-10c26f00-4437-11eb-875d-fbadf1a260fc.png)

With word boundaries and more checks:
![image](https://user-images.githubusercontent.com/1462701/102866192-29cb2000-4437-11eb-9155-d36875b56a29.png)